MekHQ/src/mekhq/gui/dialog/MekHQUnitSelectorDialog.java

```diff
@@ -106,64 +118,78 @@ public void updateOptionValues() {
-                buttonSelectClose.setName("buttonAddGM");
-                buttonSelectClose.addActionListener(this);
-                buttonSelectClose.setEnabled(false);
-                panelButtons.add(buttonSelectClose, new GridBagConstraints());
+                // This is only displayed in GM mode.
+                buttonAddGM.setText(Messages.getString("MekSelectorDialog.AddGM"));
+                buttonAddGM.setName("buttonAddGM");
+                buttonAddGM.addActionListener(evt -> addGM());
+                buttonAddGM.setEnabled(false);
+                panelButtons.add(buttonAddGM, new GridBagConstraints());
             }
 
-            // This closes the dialog
+            // This closes the dialog. Should always be around.
             buttonClose = new JButton(Messages.getString("Close"));
             buttonClose.setName("buttonClose");
             buttonClose.addActionListener(this);
         } else {
-            buttonSelectClose = new JButton(Messages.getString("MekSelectorDialog.Add"));
-            buttonSelectClose.setName("buttonAdd");
+            // This branch is for adding units where they will not be going to the hanger.
+            buttonSelect.setText(Messages.getString("MekSelectorDialog.Add"));
+            buttonSelect.setName("buttonAdd");
             //the actual work will be done by whatever called this
-            buttonSelectClose.addActionListener(evt -> setVisible(false));
-            panelButtons.add(buttonSelectClose, new GridBagConstraints());
+            buttonSelect.addActionListener(evt -> select(campaign.isGM()));
+            buttonSelect.setEnabled(true);
+            panelButtons.add(buttonSelect, new GridBagConstraints());
 
-            // This closes the dialog
-            buttonClose = new JButton(Messages.getString("Cancel"));
+            // This also closes the dialog. Different name in this state, though.
+            buttonClose.setText(Messages.getString("Cancel"));
             buttonClose.setName("buttonCancel");
             buttonClose.addActionListener(evt -> {
                 selectedUnit = null;
                 setVisible(false);
             });
         }
+        buttonClose.setEnabled(true);
         panelButtons.add(buttonClose, new GridBagConstraints());
 
-        // This displays the BV of the selected unit
-        buttonShowBV = new JButton(Messages.getString("MekSelectorDialog.BV"));
+        // This displays the BV of the selected unit.
+        buttonShowBV.setText(Messages.getString("MekSelectorDialog.BV"));
         buttonShowBV.setName("buttonShowBV");
         buttonShowBV.addActionListener(this);
         panelButtons.add(buttonShowBV, new GridBagConstraints());
 
         return panelButtons;
     }
 
-    @Override
-    protected void select(boolean isGM) {
+    /**
+     * This function validates that we have a good unit for adding to the campaign.
+     */
+    private boolean isBadSelection() {
```

Comment:
The name of the function should make it clear what it does, this doesnt. If what this does is evaluate if we can buy a unit and if that unit is supported in the campaing then you should be naming it "validatePlayableSelectedUnit"